tests/Feature/CreateThreadsTest.php

```diff
@@ -155,6 +155,23 @@ function authorized_users_can_delete_threads()
         $this->assertEquals(0, Activity::count());
     }
 
+    /** @test */
+    function a_new_thread_cannot_be_created_in_an_archived_channel()
+    {
+        $channel = factory('App\Channel')->create();
+
+        $response = $this->publishThread(['title' => 'Some Title', 'body' => 'Some body.']);
+
+        $this->get($response->headers->get('Location'))
+            ->assertSee('Some Title')
+            ->assertSee('Some body.');
+
+        $channel->archive();
+
+        $this->publishThread(['channel_id' => $channel->id])
+            ->assertSessionHasErrors('channel_id');
+    }
```

Comment:
I would drop lines 162 through 168. The `a_thread_requires_a_valid_channel` test pretty well covers this case already. Test can be as simple as:

```php
$channel = create('App\Channel', ['archived' => true]);

$this->publishThread(['channel_id' => $channel->id])
     ->assertSessionHasErrors('channel_id');
```